noodles-editor/src/noodles/operators.ts

```diff
@@ -5014,6 +5014,136 @@ export class MaskExtensionOp extends Operator<MaskExtensionOp> {
   }
 }
 
+type TimeSeriesDataPoint = {
+  time: number
+  [key: string]: number
+}
+
+/**
+ * Interpolate time series data at a given time using linear interpolation
+ */
+function interpolateTimeSeries(
+  timeSeries: TimeSeriesDataPoint[],
+  currentTime: number
+): Record<string, number> {
+  if (!timeSeries || timeSeries.length === 0) {
+    return { time: currentTime }
+  }
+
+  // If only one data point, return it
+  if (timeSeries.length === 1) {
+    return { ...timeSeries[0] }
+  }
+
+  // Sort by time to ensure proper interpolation
+  const sortedData = [...timeSeries].sort((a, b) => a.time - b.time)
```

Comment:
Is there a way that we can assume this data is sorted, or give you an option to sort it ahead of time? That way we can use a binary search which will be much faster